webapp/src/index.tsx

```diff
@@ -302,15 +303,13 @@ export default class Plugin {
                 },
                 timestamp: ev.data.timestamp,
             };
-            store.dispatch({
-                type: VOICE_CHANNEL_USER_REACTION,
-                data: {
-                    channelID: ev.broadcast.channel_id,
-                    userID: ev.data.userID,
-                    reaction,
-                },
-            });
-            setTimeout(() => {
+
+            // Clear the previous timeout of the users's reaction, if any
+            const userState = voiceUserStatus(store.getState(), ev.broadcast.channel_id, ev.data.userID);
+            clearTimeout(userState?.reaction?.timeoutID);
+
```

Comment:
I'm not sure if I'm getting this, is this removing the previous timout and replacing it with a new one per user?
so, if a user clicks like crazy, it will keep the reactions in the stream for longer as the timeouts are cleared, right? probably wont even clear unless they go outside of the queue/view

Reply:
I think the idea is when the final timeout occurs, it will remove all reactions with timestamp less than the final one, so all of them will be removed. But I think yes, while the user is clicking, the old reactions are kept in the queue. Only after the user stops clicking + 5s will all the reactions be removed.

Reply:
Hmm, maybe there is a problem if the user is reacting every 4 seconds assuming the timeout is for 5 seconds. In that case the user is not clicking like crazy but it will still clear the timeout every time, making the reactions stay in the queue...

Reply:
Thanks for your questions and comments, @Willyfrog and @cyrilzhang-mm. Thanks to that I realized this does not work at all :facepalm: It only removes the notifications of a user when the last timeout fires, of course.